### ISSUE TITLE
+layout.server.js not correctly passing data down during client-side hydration

### ISSUE THREAD

**Issue by robertadamsonsmith** [sveltejs/kit#5904]
### Describe the bug

After client side navigation, calling `await parent()` from within a +page.server.js exported load function returns `{"data":null}` instead of the value returned from the +layout.server.js exported load function

### Reproduction

- Load https://stackblitz.com/edit/sveltejs-kit-template-default-8az3ew?file=src/routes/+page.server.js
- Note that layout data, and page data are correct initially
- Click on 'Another page' link, to do a client side navigation to a new page
- Note that layout data is now incorrect (as is the layout data that was merged into page data)

### System Info

```shell
stackblitz default template fork
```

### Severity

serious, but I can work around it

**Comment by pilcrowOnPaper** [comment]
Seems like it works fine if `+layout.server.js` is changed to `+layout.js`

**Comment by robertadamsonsmith** [comment]
> Seems like it works fine if `+layout.server.js` is changed to `+layout.js`

Good to know that the issue doesn't affect `+layout.js`. I need to use `+layout.server.js` to fetch server side data though, so that doesn't help.

If I didn't need to fetch server side data in my layout, that would be fine. Unfortunately `+layout.js` doesn't run on the server, and so can't access databases etc., which is why `+layout.server.js` exists.

**Comment by rmunn** [comment]
On first navigation, the load function in `+layout.server.js` is run, because the server is hit. The second navigation is a client-side-only nagivation, so the server isn't hit, and therefore `+layout.server.js` isn't run and the values it returns aren't preserved. And instead, `await parent()` gets `{ data: null }`, which indeed doesn't seem right.

In https://github.com/sveltejs/kit/discussions/5883#discussioncomment-3393943, Rich Harris wrote that `load` results are re-used; the question I don't yet know the answer to is whether `load` results from *server* load functions are intended to be preserved after client-side navigations. I suspect the answer is yes, in which case the issue title should probably be changed to "load() results from server discarded after client-side nagivation" to better reflect what the actual bug is.

**Comment by robertadamsonsmith** [comment]
I agree - I don't know if changing the title of the issue would confuse somebody that has already seen it once though. Unless the issue extends beyond the `+layout.server.js` though, perhaps the title should be: "load() results from +layout.server.js discarded after client-side navigation"

I've reduced the reproduction to one which shows the problem in a more minimal way (i.e. without a `+page.server.js` and `await parent()`):
https://stackblitz.com/edit/sveltejs-kit-template-default-euznnp?file=src/routes/+layout.svelte

**Comment by rmunn** [comment]
https://github.com/sveltejs/kit/issues/5921#issuecomment-1216518882 suggests that it would be good to change the title. Though that will hopefully become a moot point once #5916 is merged and released.
